ragna/core/_rag.py

```diff
@@ -147,66 +211,33 @@ def __init__(
         self.params = params
         self._unpacked_params = self._unpack_chat_params(params)
 
-        self._prepared = False
         self._messages: list[Message] = []
 
-    async def prepare(self) -> Message:
```

Comment:
IMO the chat should keep the `prepare` method so users that just have local documents do not ever interact with the `Corpus` abstraction. Basically this would just be

```python
async def prepare(self):
    return await self.corpus.prepare()
```

Plus, by keeping this, we could also keep the functionality of adding the welcome message here.